functions/consumer/twitter-consumer/src/main/java/org/springframework/cloud/fn/consumer/twitter/status/update/TwitterUpdateConsumerConfiguration.java

```diff
@@ -0,0 +1,125 @@
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+/**
+ *
+ * @author Christian Tzolov
+ */
+@Configuration
+@EnableConfigurationProperties(TwitterUpdateConsumerProperties.class)
+@Import(TwitterConnectionConfiguration.class)
+public class TwitterUpdateConsumerConfiguration {
+
+	private static final Log logger = LogFactory.getLog(TwitterUpdateConsumerConfiguration.class);
+
+	@Autowired
+	public void setInfoProperties(ConfigurableEnvironment env) {
+		Properties props = new Properties();
+		props.put("spring.cloud.stream.function.definition", "toText|upper|sink");
+		env.getPropertySources().addFirst(new PropertiesPropertySource("function-dsl-props", props));
+	}
+
+	@Bean
+	public Consumer<StatusUpdate> updateStatus(Twitter twitter) {
+		return statusUpdate -> {
+			try {
+				Status status = twitter.updateStatus(statusUpdate);
+
+				if (logger.isDebugEnabled()) {
+					logger.debug(status);
+				}
+			}
+			catch (TwitterException e) {
+				logger.error("Failed apply update status: " + statusUpdate, e);
+			}
+		};
+	}
+
+	@Bean
+	public Function<Message<?>, StatusUpdate> toStatusUpdateQuery(TwitterUpdateConsumerProperties updateProperties) {
+
+		return message -> {
+
+			String updateText = updateProperties.getText().getValue(message, String.class);
+
+			StatusUpdate statusUpdate = new StatusUpdate(updateText);
+
+			if (updateProperties.getAttachmentUrl() != null) {
+				statusUpdate.setAttachmentUrl(updateProperties.getAttachmentUrl().getValue(message, String.class));
+			}
+
+			if (updateProperties.getPlaceId() != null) {
+				statusUpdate.setPlaceId(updateProperties.getPlaceId().getValue(message, String.class));
+			}
+
+			if (updateProperties.getInReplyToStatusId() != null) {
+				statusUpdate.setInReplyToStatusId(updateProperties.getInReplyToStatusId().getValue(message, int.class));
+				statusUpdate.setAutoPopulateReplyMetadata(true);
+			}
+
+			if (updateProperties.getDisplayCoordinates() != null) {
+				statusUpdate.setDisplayCoordinates(
+						updateProperties.getDisplayCoordinates().getValue(message, boolean.class));
+			}
+
+			if (updateProperties.getMediaIds() != null) {
+				long[] mediaIds = updateProperties.getMediaIds().getValue(message, long[].class);
+				statusUpdate.setMediaIds(mediaIds);
+			}
+
+			if (updateProperties.getLocation().getLat() != null) {
+				Assert.notNull(updateProperties.getLocation().getLon(),
+						"If the latitude is set then the longitude must be set too");
+				double lat = updateProperties.getLocation().getLat().getValue(message, Double.class);
+				double lon = updateProperties.getLocation().getLon().getValue(message, Double.class);
+				statusUpdate.setLocation(new GeoLocation(lat, lon));
```

Comment:
Can this be a validation rule in the properties?

Reply:
It is already in the [property validations](https://github.com/spring-cloud/stream-applications/blob/master/functions/consumer/twitter-consumer/src/main/java/org/springframework/cloud/fn/consumer/twitter/status/update/TwitterUpdateConsumerProperties.java#L138-L142). I guess i forgot to clean it from the code.